### PR TITLE
fix: store signed blocks instead of regular blocks.

### PR DESCRIPTION
**Pull request by mpaulucci**
**Motivation**
We need to store signed blocks since we will need to pass them to other peers on request.

### CODE REVIEW

File: lib/lambda_ethereum_consensus/store/block_store.ex [hunk 1]
```diff
@@ -8,13 +8,14 @@ defmodule LambdaEthereumConsensus.Store.BlockStore do
...
     {:ok, block_root} = Ssz.hash_tree_root(block)
-    {:ok, encoded_block} = Ssz.to_ssz(block)
+    {:ok, encoded_signed_block} = Ssz.to_ssz(signed_block)
 
     key = block_key(block_root)
-    Db.put(key, encoded_block)
+    Db.put(key, encoded_signed_block)
 
     # WARN: this overrides any previous mapping for the same slot
     # TODO: this should apply fork-choice if not applied elsewhere
```
Comment (Godspower-Eze) [inline lib/lambda_ethereum_consensus/store/block_store.ex:21]: I think `block_root_by_slot_key(block.slot)` should change to `block_root_by_slot_key(block.message.slot)`.

File: lib/lambda_ethereum_consensus/fork_choice/store.ex [hunk 2]
```diff
@@ -50,10 +50,10 @@ defmodule LambdaEthereumConsensus.ForkChoice.Store do
   ##########################
 
   @impl GenServer
-  @spec init({BeaconState.t(), BeaconBlock.t()}) :: {:ok, Store.t()} | {:stop, any}
-  def init({anchor_state = %BeaconState{}, anchor_block = %BeaconBlock{}}) do
+  @spec init({BeaconState.t(), SignedBeaconBlock.t()}) :: {:ok, Store.t()} | {:stop, any}
+  def init({anchor_state = %BeaconState{}, signed_anchor_block = %SignedBeaconBlock{}}) do
     result =
-      case Helpers.get_forkchoice_store(anchor_state, anchor_block) do
+      case Helpers.get_forkchoice_store(anchor_state, signed_anchor_block.message) do
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/fork_choice/store.ex:56]: I think we should also store the signed version in fork choice.

File: lib/lambda_ethereum_consensus/store/block_store.ex (hunk 1 above)
Comment (mpaulucci) [reply lib/lambda_ethereum_consensus/store/block_store.ex:21]: But block is not a signed block in this case, see line 13.

File: lib/lambda_ethereum_consensus/fork_choice/store.ex (hunk 2 above)
Comment (mpaulucci) [reply lib/lambda_ethereum_consensus/fork_choice/store.ex:56]: What would be the benefit? BTW, the Store as defined in the spec, stores blocks, not signed blocks: https://eth2book.info/capella/annotated-spec/#store

Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/fork_choice/store.ex:56]: That's interesting. I was thinking we might want to validate block signatures in the fork choice.

Comment (mpaulucci) [reply lib/lambda_ethereum_consensus/fork_choice/store.ex:56]: We do, we get `signed_block` in the `on_block` handler, then we do the validation and we store just the block

**Pull request merged**